pkg/apis/pipeline/v1beta1/resource_types.go

```diff
@@ -70,6 +70,16 @@ type TaskResources struct {
 	// DeclaredPipelineResources to the input PipelineResources required by the Task.
 	// +listType=atomic
 	Outputs []TaskResource `json:"outputs,omitempty"`
+	// Limits describes the maximum amount of compute resources allowed.
```

Comment:
We'll want to document this feature really well, because it's not intuitive how these resource requirements will be applied. I'd suggest adding a bit of detail here, and a lot of detail in [docs/compute-resources.md](https://github.com/tektoncd/pipeline/blob/main/docs/compute-resources.md).